src/core/text/Text.js

```diff
@@ -175,42 +143,17 @@ Object.defineProperties(Text.prototype, {
         set: function (style)
         {
             style = style || {};
-
-            if (typeof style.fill === 'number') {
-                style.fill = utils.hex2string(style.fill);
-            }
-
-            if (typeof style.stroke === 'number') {
-                style.stroke = utils.hex2string(style.stroke);
-            }
-
-            if (typeof style.dropShadowColor === 'number') {
-                style.dropShadowColor = utils.hex2string(style.dropShadowColor);
+            if (style instanceof TextStyle)
+            {
+                this._style = style;
+            } else
+            {
+                this._style = new TextStyle(style);
             }
-
-            style.font = style.font || 'bold 20pt Arial';
-            style.fill = style.fill || 'black';
-            style.align = style.align || 'left';
-            style.stroke = style.stroke || 'black'; //provide a default, see: https://github.com/pixijs/pixi.js/issues/136
-            style.strokeThickness = style.strokeThickness || 0;
-            style.wordWrap = style.wordWrap || false;
-            style.wordWrapWidth = style.wordWrapWidth || 100;
-
-            style.dropShadow = style.dropShadow || false;
-            style.dropShadowColor = style.dropShadowColor || '#000000';
-            style.dropShadowAngle = style.dropShadowAngle !== undefined ? style.dropShadowAngle : Math.PI / 6;
-            style.dropShadowDistance = style.dropShadowDistance !== undefined ? style.dropShadowDistance : 5;
-            style.dropShadowBlur = style.dropShadowBlur !== undefined ? style.dropShadowBlur : 0; //shadowBlur is '0' by default according to HTML
-
-            style.padding = style.padding || 0;
-
-            style.textBaseline = style.textBaseline || 'alphabetic';
-
-            style.lineJoin = style.lineJoin || 'miter';
-            style.miterLimit = style.miterLimit || 10;
-
-            this._style = style;
-            this.dirty = true;
+            this._styleListener = this._style.on('styleChanged', function ()
+            {
+                this.dirty = true;
+            }.bind(this));
```

Comment:
Couple things here:

1. `.on()` returns the object you are binding to, not the listener function
2. `.on()` can take 3 params, where the third is the context which is much faster on calls than `.bind()`

I would do this instead:

```js
Text.prototype._onStyleChange = function ()
{
    this.dirty = true;
};

// ...

this._style.on('styleChanged', this._onStyleChange, this);

// ...

this._style.off('styleChanged', this._onStyleChange, this);
```